polars/polars-core/src/chunked_array/temporal/datetime.rs

```diff
@@ -142,16 +148,19 @@ impl DatetimeChunked {
             .unwrap()
             .and_hms_opt(0, 0, 0)
             .unwrap();
-        let fmted = format!("{}", dt.format(fmt));
-
-        #[allow(unused_mut)]
-        let mut ca = self.clone();
-        #[cfg(feature = "timezones")]
-        if self.time_zone().is_some() {
-            ca = ca.cast_time_zone(Some("UTC")).unwrap();
-        }
+        let fmted = match self.time_zone() {
+            #[cfg(feature = "timezones")]
+            Some(_) => {
+                let tz: Tz = "UTC".to_string().parse().unwrap();
```

Comment:
I don't think we should allocate a string to get the UTC timezone. We probably can just take the UTC timezone from `chrono_tz`.